mixing/mixclient/client.go

```diff
@@ -1044,17 +1149,26 @@ func (c *Client) run(ctx context.Context, ps *pairedSessions, madePairing *bool)
 
 		p.ke = ke
 		p.rs = rs
-		return p.signAndSubmit(ke)
+		return nil
 	})
 	if err != nil {
-		c.logf("%v", err)
+		sesLog.logf("%v", err)
+	}
+	err = c.sendLocalPeerMsgs(ctx, sesRun, func(p *peer) mixing.Message {
+		if p.ke == nil {
```

Comment:
I see this pattern repeated for most of the messages, but, unless I'm missing something, why are these not just returning the message directly?  After all, `if p.ke == nil`, well, `p.ke` is `nil`, so return `return p.ke` == `return nil`.

Reply:
this had the potential to cause panics during message serialization (return a nil p.ke here, it gets wrapped as a non-mil mixing.Message, and the nil check in the send func won't trigger because the interface itself is non-nil).

It shouldn't happen, because either the iteration above should either set a non-nil p.ke (or ct, or sr...) *or* set p.triggeredBlame (in which case we discard what this closure here returns and use that instead).

Reply:
Ah, right, you're returning an interface, so that could theoretically happen.  As you mentioned, it shouldn't, but probably better to be safe when returning an interface.